webrender/src/render_backend.rs

```diff
@@ -261,24 +260,33 @@ impl RenderBackend {
                               .unwrap()
                         }
                         ApiMsg::RequestWebGLContext(size, attributes, tx) => {
-                            if let Some(ref handle) = self.webrender_context_handle {
-                                match GLContext::<NativeGLContext>::new(size, attributes, ColorAttachmentType::Texture, Some(handle)) {
+                            if let Some(ref wrapper) = self.webrender_context_handle {
+                                // Try to create a shared context first.
+                                let mut shared = true;
```

Comment:
Yes - your thinking is exactly right. The corresponding servo code is https://github.com/servo/servo/pull/13382/commits/925c8fca9bc1352576acde7b3ce21ea0600f382a - it just saves servo having to worry about what type of context to create.